src/nsrt_learning/strips_learning/base_strips_learner.py

```diff
@@ -0,0 +1,243 @@
+"""Base class for a STRIPS operator learning algorithm."""
+
+import abc
+from typing import FrozenSet, Iterator, List, Set, Tuple
+
+from predicators.src import utils
+from predicators.src.planning import task_plan_grounding
+from predicators.src.settings import CFG
+from predicators.src.structs import DummyOption, GroundAtom, LiftedAtom, \
+    LowLevelTrajectory, OptionSpec, PartialNSRTAndDatastore, Predicate, \
+    Segment, State, STRIPSOperator, Task, _GroundNSRT
+
+
+class BaseSTRIPSLearner(abc.ABC):
+    """Base class definition."""
+
+    def __init__(
+        self,
+        trajectories: List[LowLevelTrajectory],
+        train_tasks: List[Task],
+        predicates: Set[Predicate],
```

Comment:
why do we need the predicates passed in exactly? aren't they already in the segmented trajs?

Reply:
yeah, but we do this sort of thing in a lot of places. technically predicates are always redundant with data. i think it's fine as is. btw, the predicates are used in both harmlessness checking and backchaining (to initialize the side predicates of the most general operators)